_posts/2020-02-26-ads.md

```diff
@@ -0,0 +1,36 @@
+---
```

Comment:
```suggestion
------
layout: post
title: Monetize your ads!
---

Hi everyone!

This time I would like to share with you our experience with ads in Just Draw It. We decided to start monetizing the app using ads because of its easy setup and because we already had some experience with them. We didn't know that it wouldn't be an easy task.

## Beginning

There are two main types of ads: *Interstitial*, which is a full screen ad, and *Banner*, which is a small size ad that is rendered as any other layout component. Of course, they don't monetize the same, *interstitial* ones gives better results.

So, we decided to show an *Interstitial* ad whenever our users open a saved route. Everything was setup as we expected, but after one month Google stopped serving ads to Just Draw It 😭. We didn't know if the problem was on our side or theirs, but we didn't change our code, so... oh damn Google.

This issue was fixed after a month, but it continued to go up and down after some weeks. Thankfully, this is not happening any more 🥳

By the way, our initial implementation of ads monetization was using just one *Interstitial* when a user opened a saved route and it was going good. This approach gave us a regular income each month, but as the app was growing in users, the ads income should be growing too. This didn't happen.

## Investigation

We have almost every user interaction tracked in our analytics service provider, so we investigated the actions that a free user does the most. These actions would give us most impressions in Google ads, so which ones were them?

* Click undo
* Click close
* Click save

These actions are sorted by its usage. We discarded the save action because we already show an ad in **save** flow. Between undo and close, although undo has more usage than close we thought it wasn't a good choice. Undo is an action that the users needs to *undo* the last drawn segment of the route and usually they click on it several times in a short period of time. It wouldn't be a good user experiencie if we show an ad on each click.

So, we decided to show an *Interstitial* every time our users click on **close** route. This was risky because we didn't want bad reviews or a decrease in our downloads rate (or even worse, an increase of our uninstalls rate), so this was just an experiment. And... it works! We didn't receive any bad feedback/review, the downloads went normal and.. the most important, our ads income grew!

![Just Draw It](/public/ads.jpg)

Here it's a chart about our ads income. As you can see, in July and September Google stopped serving us ads. After September, everything started to go back to normal until our change in February that our income grew by **80%**!

Let's take risk!
```